core/VectorState.hpp

```diff
@@ -2,21 +2,67 @@
 
 #include <stdint.h>
 #include <iostream>
+#include <vector>
 
 #include "core/AtlasState.hpp"
+#include "core/VecNums.hpp"
+#include "include/AtlasUtils.hpp"
 
 namespace atlas
 {
-    class VectorState
+    constexpr uint8_t VLEN_MIN = 32;
+
+    class VectorConfig
     {
       public:
+        VectorConfig() : vlen_(64), lmul_(8), sew_(8), vta_(false), vma_(false), vl_(0), vstart_(0)
+        {
+        }
+
```

Comment:
Do not implement a default constructor.  Initialize the variables in line below and default this constructor.